nodeup/pkg/model/cilium.go

```diff
@@ -0,0 +1,197 @@
+		return nil
+	}
+
+	if err := b.builldBPFMount(c); err != nil {
+		return err
+	}
+
+	if networking.Cilium.EtcdManaged {
+		if err := b.buildCiliumEtcdSecrets(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func (b *CiliumBuilder) builldBPFMount(c *fi.ModelBuilderContext) error {
+
+	var fsdata unix.Statfs_t
+	err := unix.Statfs("/sys/fs/bpf", &fsdata)
+
+	if err != nil {
+		return fmt.Errorf("error checking for /sys/fs/bpf: %v", err)
+	}
+
+	// equivalent to unix.BPF_FS_MAGIC in golang.org/x/sys/unix
+	BPF_FS_MAGIC := uint32(0xcafe4a11)
+
+	// systemd v238 includes the bpffs mount by default; and gives an error "has a bad unit file setting" if we try to mount it again (see mount_point_is_api)
+	alreadyMounted := uint32(fsdata.Type) == BPF_FS_MAGIC
+
+	if !alreadyMounted {
+		unit := s(`
+[Unit]
+Description=Cilium BPF mounts
+Documentation=http://docs.cilium.io/
+DefaultDependencies=no
+Before=local-fs.target umount.target kubelet.service
+
+[Mount]
+What=bpffs
+Where=/sys/fs/bpf
+Type=bpf
+
+[Install]
+WantedBy=multi-user.target
+`)
+
+		service := &nodetasks.Service{
+			Name:       "sys-fs-bpf.mount",
+			Definition: unit,
+		}
+		service.InitDefaults()
+		c.AddTask(service)
+	}
+
+	return nil
+}
+
+func (b *CiliumBuilder) buildCiliumEtcdSecrets(c *fi.ModelBuilderContext) error {
+
+	if b.IsMaster {
+		d := "/etc/kubernetes/pki/etcd-manager-cilium"
+
+		keys := make(map[string]string)
+		keys["etcd-manager-ca"] = "etcd-manager-ca-cilium"
+		keys["etcd-peers-ca"] = "etcd-peers-ca-cilium"
+		keys["etcd-clients-ca"] = "etcd-clients-ca-cilium"
+
+		for fileName, keystoreName := range keys {
+			_, err := b.KeyStore.FindCert(keystoreName)
+			if err != nil {
+				return err
+			}
+
+			if err := b.BuildCertificateTask(c, keystoreName, d+"/"+fileName+".crt"); err != nil {
+				return err
+			}
+			if err := b.BuildPrivateKeyTask(c, keystoreName, d+"/"+fileName+".key"); err != nil {
+				return err
+			}
+		}
+	}
+
+	etcdClientsCACertificate, err := b.KeyStore.FindCert("etcd-clients-ca-cilium")
+	if err != nil {
+		return err
+	}
+
+	etcdClientsCAPrivateKey, err := b.KeyStore.FindPrivateKey("etcd-clients-ca-cilium")
+	if err != nil {
+		return err
+	}
+
+	dir := "/etc/kubernetes/pki/cilium"
+
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("error creating directories %q: %v", dir, err)
+	}
+
+	{
+		p := filepath.Join(dir, "etcd-ca.crt")
+		certBytes := pkiutil.EncodeCertPEM(etcdClientsCACertificate.Certificate)
+		if err := ioutil.WriteFile(p, certBytes, 0644); err != nil {
+			return fmt.Errorf("error writing certificate key file %q: %v", p, err)
+		}
+	}
+
+	name := "etcd-client"
+
+	humanName := dir + "/" + name
+	privateKey, err := pkiutil.NewPrivateKey()
+	if err != nil {
+		return fmt.Errorf("unable to create private key %q: %v", humanName, err)
+	}
+	privateKeyBytes := pkiutil.EncodePrivateKeyPEM(privateKey)
+
+	certConfig := &certutil.Config{
+		CommonName: "kube-apiserver",
```

Comment:
looks like a copy/paste from [etcd_manager_tls.go](https://github.com/kubernetes/kops/blob/master/nodeup/pkg/model/etcd_manager_tls.go), should this cert have a different CN?